org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/exporter/AbstractVelocityExporter.java

```diff
@@ -0,0 +1,148 @@
+            Velocity.setProperty("classpath.resource.loader.class", ClasspathResourceLoader.class.getName());
+
+            Velocity.init();
+
+            final VelocityContext context = createVelocityContext(outputResult, reportConfiguration);
+            final Template template = getTemplate(reportConfiguration);
+
+            try (final StringWriter sw = new StringWriter();) {
+                if (template != null) {
+                    template.merge(context, sw);
+                }
+                bw.write(sw.getBuffer().toString());
+            }
+        }
+    }
+
+    private VelocityContext createVelocityContext(final OutputResult outputResult,
+                                                  final ReportConfiguration reportConfiguration) {
+        final VelocityContext context = new VelocityContext();
+        final List<Archive> archiveFiles = getSortedArchives(outputResult);
+        context.put("archiveFiles", archiveFiles);
+
+        context.put("sourcePath", outputResult.getFinderResult().getScanDirectory().getAbsolutePath());
+        context.put("detectionStatusStatistics", outputResult.getDetectionStatusStatistics());
+        context.put("legalStatusStatistics", outputResult.getLegalStatusStatistics());
+        context.put("generalStatistics", outputResult.getGeneralStatistics());
+        context.put("messageDigestAlgorithm", outputResult.getMessageDigestAlgorithm());
+        context.put("reportConfiguration", reportConfiguration);
+        context.put("pomResolutionUsed", outputResult.isPomResolutionUsed());
+
+        final List<License> distinctLicenses = collectDistinctLicenses(archiveFiles);
+        context.put("distinctLicenses", distinctLicenses);
+
+        additionalSetup(context, outputResult);
+        return context;
+    }
+
+    private List<Archive> getSortedArchives(final OutputResult outputResult) {
+        final List<Archive> archiveFiles = outputResult.getFinderResult().getArchiveFiles();
+        Collections.sort(archiveFiles);
+        return archiveFiles;
+    }
+
+    private List<License> collectDistinctLicenses(final List<Archive> archiveFiles) {
+        final Set<License> allLicenses = new HashSet<>();
+        for (final Archive archive : archiveFiles) {
+            allLicenses.addAll(archive.getLicenses());
+        }
+        final List<License> distinctLicenses = new ArrayList<>(allLicenses);
+
+        Collections.sort(distinctLicenses);
+        return distinctLicenses;
+    }
+
+    /**
+     * @param context
+     * @param outputResult
+     */
+    protected void additionalSetup(VelocityContext context, OutputResult outputResult) {
+        // EMPTY
+    }
+
+    /**
+     * Obtains the velocity template to use in the export method.
+     * 
+     * @return the velocity template to use
+     */
+    protected Template getTemplate(final ReportConfiguration reportConfiguration) {
+        if (reportConfiguration.getTemplateFile() != null) {
```

Comment:
[Optional] Substitute these lines with:
`return (reportConfiguration.getTemplateFile == null) ? getDefaultTemplate() : Velocity.getTemplate(reportConfiguration.getTemplateFile().getAbsolutePath());`

Reply:
We do not have clear rules on that.
ABS rules advise against using the ternary operator '?' if the expression is so complex that it has to be distrubuted on multiple lines. This is because it is hard to read and explicit if-else statements are more clear and therefore less prone to errors.